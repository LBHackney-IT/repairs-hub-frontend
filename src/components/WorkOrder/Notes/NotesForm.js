import PropTypes from 'prop-types'
import { useForm } from 'react-hook-form'
import { TextArea, PrimarySubmitButton } from '../../Form'
import { buildNoteFormData } from '../../../utils/hact/job-status-update/notes-form'

const NotesForm = ({
  onFormSubmit,
  tabName,
  workOrderReference,
  displayForm,
  showForm,
}) => {
  const { register, handleSubmit, errors } = useForm()

  const onSubmit = async (formData) => {
    const notesFormData = buildNoteFormData(formData)

    onFormSubmit(notesFormData)
  }

  return (
    <>
      <div className="govuk-!-margin-bottom-6">
        <h2 className="lbh-heading-h2 display-inline govuk-!-margin-right-6">
          {tabName}
        </h2>
        {!displayForm && (
          <div className="display-inline">
            <a
              onClick={showForm}
              href="#"
              className="repairs-hub-link lbh-body-s"
            >
              Add a new note
            </a>
          </div>
        )}
      </div>
      <div className="govuk-grid-row">
        <div className="govuk-grid-column-two-thirds">
          {displayForm && (
            <form role="form" id="notes-form" onSubmit={handleSubmit(onSubmit)}>
              <TextArea
                name="note"
                label="Note"
                placeholder="Write a new note..."
                required={true}
                register={register({
                  required: 'Please enter a note',
                  maxLength: {
                    value: 2000,
                    message:
                      'You have exceeded the maximum amount of 2000 characters',
                  },
                })}
                error={errors && errors.note}
              />
              <input
                id="workOrderReference"
                name="workOrderReference"
                label="workOrderReference"
                type="hidden"
                value={workOrderReference}
                ref={register}
              />

              <PrimarySubmitButton label="Publish note" />
            </form>
          )}
        </div>
      </div>
    </>
  )
}

NotesForm.propTypes = {
  onFormSubmit: PropTypes.func.isRequired,
  tabName: PropTypes.string.isRequired,
  workOrderReference: PropTypes.string.isRequired,
  displayForm: PropTypes.bool.isRequired,
  showForm: PropTypes.func.isRequired,
}

export default NotesForm
