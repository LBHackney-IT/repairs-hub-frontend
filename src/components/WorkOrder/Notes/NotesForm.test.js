import { render } from '@testing-library/react'
import NotesForm from './NotesForm'

describe('NotesForm component', () => {
  const props = {
    tabName: 'notes',
    workOrderReference: '10002222',
    onFormSubmit: jest.fn(),
    showForm: jest.fn(),
  }

  it('should render properly when display form is false', () => {
    const { asFragment } = render(
      <NotesForm
        tabName={props.tabName}
        workOrderReference={props.workOrderReference}
        onFormSubmit={props.onFormSubmit}
        showForm={props.showForm}
        displayForm={false}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render properly when display form is true', () => {
    const { asFragment } = render(
      <NotesForm
        tabName={props.tabName}
        workOrderReference={props.workOrderReference}
        onFormSubmit={props.onFormSubmit}
        showForm={props.showForm}
        displayForm={true}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
