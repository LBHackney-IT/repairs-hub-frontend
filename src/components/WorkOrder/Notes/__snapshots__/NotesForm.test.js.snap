// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesForm component should render properly when display form is false 1`] = `
<DocumentFragment>
  <div
    class="govuk-!-margin-bottom-6"
  >
    <h2
      class="lbh-heading-h2 display-inline govuk-!-margin-right-6"
    >
      notes
    </h2>
    <div
      class="display-inline"
    >
      <a
        class="lbh-link"
        href="#"
      >
        Add a new note
      </a>
    </div>
  </div>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-two-thirds"
    />
  </div>
</DocumentFragment>
`;

exports[`NotesForm component should render properly when display form is true 1`] = `
<DocumentFragment>
  <div
    class="govuk-!-margin-bottom-6"
  >
    <h2
      class="lbh-heading-h2 display-inline govuk-!-margin-right-6"
    >
      notes
    </h2>
  </div>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-two-thirds"
    >
      <form
        id="notes-form"
        role="form"
      >
        <div
          class="govuk-form-group lbh-form-group"
          id="note-form-group"
        >
          <label
            class="govuk-label lbh-label"
            for="note"
          >
            Note 
            <span
              class="govuk-required"
            >
              *
            </span>
          </label>
          <textarea
            aria-describedby="note-hint note-error"
            class="govuk-textarea lbh-textarea"
            id="note"
            name="note"
            placeholder="Write a new note..."
            rows="4"
          />
        </div>
        <input
          id="workOrderReference"
          label="workOrderReference"
          name="workOrderReference"
          type="hidden"
          value="10002222"
        />
        <div
          class="govuk-form-group lbh-form-group"
        >
          <button
            class="govuk-button lbh-button"
            data-module="govuk-button"
            type="submit"
          >
            Publish note
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
